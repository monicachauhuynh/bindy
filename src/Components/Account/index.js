import React, { useState } from "react";
import {
  Form,
  Checkbox,
  Dropdown,
  Button,
  Input,
  Select,
} from "semantic-ui-react";
import "./style.css";
import {
  genderOptions,
  interestOptions,
  keywords,
  cities,
  sexualOrientaions,
} from "./constants";

const Account = () => {
  const [keywordOptions, setKeywordOptions] = useState(keywords);
  const [locationOptions, setLocationOptions] = useState(cities);
  const [formData, setFormData] = useState({});

  const handleSubmit = (event) => {
    event.preventDefault();
    const requestOptions = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(formData),
    };
    fetch("http://localhost:5000/user", requestOptions).then((response) =>
      response.json()
    );
  };

  const handleAddition = (e, result) => {
    const { id, value } = result;
    if (id === "keywords") {
      setKeywordOptions([{ text: value, value }, ...keywordOptions]);
    } else if (id === "location") {
      setLocationOptions([{ text: value, value }, ...locationOptions]);
    }
  };

  const handleInputChange = (e) => {
    e.preventDefault();
    setFormData({
      ...formData,
      [e.target.id]: e.target.value,
    });
  };

  const handleDropdownChange = (e, result) => {
    e.preventDefault();
    const { id, value } = result;
    setFormData({
      ...formData,
      [id]: value,
    });
  };

  return (
    <div className="wrapper">
      <div className="heading">
        <div className="title">Edit Profile</div>
        <div className="button-container">
          <Button className="save-btn" onClick={handleSubmit}>
            SAVE
          </Button>
          <Button className="exit-btn">EXIT</Button>
        </div>
      </div>
      <Form className="account-form">
        <div className="box left-box">
          <Checkbox
            toggle
            label="Show my Bindy profile"
            className="toggle-btn"
          />
          <textarea
            id="about"
            label="About"
            placeholder="Tell us more about you..."
            className="about-field"
            onChange={handleInputChange}
          />
          <Form.Input
            fluid
            id="employer"
            label="School/Company"
            placeholder="School/Company name"
            onChange={handleInputChange}
          />

          <Form.Field>
            <label>Keywords</label>
            <Dropdown
              options={keywordOptions}
              placeholder="Add keywords about yourself"
              search
              selection
              fluid
              multiple
              allowAdditions
              onAddItem={handleAddition}
              onChange={handleDropdownChange}
              id="keywords"
              className="keywords-input"
            />
          </Form.Field>
        </div>

        <div className="box right-box">
          <Form.Field inline>
            <label>Pronouns</label>
            <Input
              placeholder="they/them"
              onChange={handleInputChange}
              id="pronouns"
            />
          </Form.Field>

          <Form.Field inline>
            <label>Gender</label>
            <Select
              className="select"
              options={genderOptions}
              placeholder="Gender"
              onChange={handleDropdownChange}
              id="gender"
            />
          </Form.Field>

          <Form.Field inline>
            <label>Interest</label>
            <Select
              className="select"
              options={interestOptions}
              placeholder="Everyone"
              id="interest"
              onChange={handleDropdownChange}
            />
          </Form.Field>

          <Form.Field>
            <label>Sexual Orientation</label>
            <Select
              fluid
              placeholder="Pansexual"
              options={sexualOrientaions}
              className="select"
              //TODO - handle sexual orientation change
              //   id="interest"
              // onChange={handleDropdownChange}
            />
          </Form.Field>

          <Form.Field>
            <label>City</label>
            <Dropdown
              placeholder="Search for your city name"
              className="select"
              search
              selection
              options={locationOptions}
              id="location"
              onChange={handleDropdownChange}
              allowAdditions
              onAddItem={handleAddition}
            />
          </Form.Field>

          <button fluid basic color="black" className="connect-btn">
            CONNECT INSTAGRAM
          </button>
          <button fluid basic color="black" className="connect-btn">
            CONNECT SPOTIFY
          </button>
        </div>
      </Form>
    </div>
  );
};

export default Account;
