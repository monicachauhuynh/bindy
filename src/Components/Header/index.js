import React from "react";
import Logo from "./Logo";
import Menu from "./Menu";
import styled from "styled-components";

const NavBar = styled.div`
  display: flex;
  align-items: center;
  justify-content: start;
  padding: 20px;
`;

const Header1 = () => {
  return (
    <NavBar>
      <Logo />
      <Menu />
    </NavBar>
  );
};

export default Header1;
