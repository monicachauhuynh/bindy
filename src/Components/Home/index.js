import React from "react";

import logo from "../../Assets/bindy-logo.png";

const Home = () => {
  return (
    <div class="ui segment">
      <img class="ui centered medium image" src={logo} alt="Logo" />
    </div>
  );
};

export default Home;