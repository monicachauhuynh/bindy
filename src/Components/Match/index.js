import React from "react";
import UserCard from "./UserCard";
import { DummyUser } from "./const";

const Match = () => {
  return <UserCard user={DummyUser} />;
};

export default Match;
