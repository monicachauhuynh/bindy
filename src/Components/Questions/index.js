import React from "react";
import { Box } from "./style";
import { Container, Button, Header } from "semantic-ui-react";
import { ReactComponent as QuizImg } from "../../Assets/quiz_img.svg";

const Questions = () => {
  return (
    <Container>
      <Box>
        <Header className="box-title">36 Questions Game</Header>
        <p>
          This text will explain how the game works. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Nunc, nullam ut suscipit diam tristique
          egestas amet. Sit. Lorem ipsum dolor sit amet, consectetur adipiscing
          elit. Nunc, nullam ut suscipit diam tristique egestas amet. Nunc,
          nullam ut suscipit.
        </p>
        <QuizImg />
        <Button className="start-btn">START</Button>
      </Box>
    </Container>
  );
};

export default Questions;
