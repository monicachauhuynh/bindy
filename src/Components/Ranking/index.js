import React, { useState } from "react";
import { DoneButton, Footer, CardList, Header } from "./style";
import { Container } from "semantic-ui-react";
import MatchCard from "./components/MatchCard";

const mockData = [
  { name: "Bob 1", distance: "12 km away", keywords: ["dog", "cat"], id: "1" },
  { name: "Two", distance: "12 km away", keywords: ["dog", "cat"], id: "2" },
  { name: "Three", distance: "12 km away", keywords: ["dog", "cat"], id: "3" },
];

const Questions = () => {
  const [items, setItems] = useState([]);

  const removeItem = (item) => {
    setItems((prevState) => prevState.filter((prevItem) => prevItem !== item));
  };

  const handleSelectCard = (card) => {
    console.log(card);
    setItems([...items, card]);
    console.log(items);
  };

  const handleDeselectCard = (card) => {
    removeItem(card);
  };

  return (
    <Header>
      <h1>Select the matches that you want to keep in contact </h1>
      <hr />
      <CardList>
        {mockData &&
          mockData.map((card) => (
            <MatchCard
              name={card.name}
              // handleAddItem={handleAddItem}
              onSelect={handleSelectCard}
              onDeselect={handleDeselectCard}
            />
          ))}
      </CardList>
      <Footer>
        <h1>
          {items.length}/{mockData.length} selected
        </h1>
        <DoneButton>DONE</DoneButton>
      </Footer>
    </Header>
  );
};

export default Questions;
