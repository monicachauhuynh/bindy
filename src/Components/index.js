export { default as Signup } from './Signup';
export { default as Signin } from './Signin';
export { default as Header } from './Header';
export { default as Home } from './Home';
export { default as Account } from './Account';
export { default as Questions } from './Questions';
export { default as Messages } from './Messages';
export { default as Match } from './Match';