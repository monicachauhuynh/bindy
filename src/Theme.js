import React from "react";
import { ThemeProvider } from "styled-components";

const theme = {
    colors: {
        lightPurple: "#C4B5F9",
        darkPurple: "8B5CB4",
        lightGreen: "#CAF4ED",
        lightGray: "#E5E5E5",
        darkGray: "#3e4347"
    },
    linearGradient: `linear-gradient(
        90deg,
        #caf4ed -18.14%,
        rgba(196, 181, 249, 0.8) 95.7%
      )`
};

const Theme = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;