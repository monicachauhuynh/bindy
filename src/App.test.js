import React from "react";
import renderer from "react-test-renderer";
import { Signup } from "./Components";
import { theme } from "./Theme";
import { ThemeProvider } from "styled-components";

test("Renders Sign up", () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Signup />
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
