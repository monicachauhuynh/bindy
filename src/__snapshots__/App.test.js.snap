// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Sign up 1`] = `
<div
  style={
    Object {
      "marginLeft": "140px",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "40px",
      }
    }
  >
    <h1
      className="sc-bdvvtL bpnWpV"
    >
      Sign Up
    </h1>
  </div>
  <div
    className="ui container grid"
  >
    <form
      className="ui form sc-gsDKAQ eHpZrV"
      onSubmit={[Function]}
    >
      <div
        className="seven wide computer sixteen wide mobile eight wide tablet column"
      >
        <div
          className="inline field"
        >
          <label>
            Username
          </label>
          <div
            className="ui input"
          >
            <input
              id="username"
              name="username"
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="inline field"
        >
          <label>
            Name
          </label>
          <div
            className="ui input"
          >
            <input
              id="name"
              name="name"
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="inline field"
        >
          <label>
            D. O. B
          </label>
          <div
            className="ui input"
          >
            <input
              id="dob"
              name="dob"
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="inline field"
        >
          <label>
            Gender
          </label>
          <div
            aria-expanded={false}
            className="ui selection dropdown select"
            id="gender"
            name="gender"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              className="divider default text"
              role="alert"
            >
              Male
            </div>
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={false}
                aria-selected={true}
                className="selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Male
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Female
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Other
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="inline field"
        >
          <label>
            Interest
          </label>
          <div
            aria-expanded={false}
            className="ui selection dropdown select"
            id="interest"
            name="interest"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              className="divider default text"
              role="alert"
            >
              Everyone
            </div>
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={false}
                aria-selected={true}
                className="selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Male
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Female
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Everyone
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Max Distance
          </label>
          <div
            className="signup-slider"
            style={
              Object {
                "flexBasis": "calc(100% - 20px)",
                "height": "45px",
                "marginLeft": "10px",
                "marginRight": "10px",
                "position": "relative",
              }
            }
          >
            <div
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "backgroundColor": "rgb(150, 150, 150)",
                  "borderRadius": "5px",
                  "height": "10px",
                  "marginTop": "18px",
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <div
              className="slider-handles"
            >
              <div
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": "50%",
                    "color": "#333",
                    "cursor": "pointer",
                    "height": "20px",
                    "left": "0%",
                    "marginLeft": "-10px",
                    "marginTop": "12px",
                    "position": "absolute",
                    "textAlign": "center",
                    "width": "20px",
                    "zIndex": 2,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "fontSize": "14px",
                      "marginTop": "-20px",
                    }
                  }
                >
                  0
                </div>
              </div>
              <div
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": "50%",
                    "color": "#333",
                    "cursor": "pointer",
                    "height": "20px",
                    "left": "100%",
                    "marginLeft": "-10px",
                    "marginTop": "12px",
                    "position": "absolute",
                    "textAlign": "center",
                    "width": "20px",
                    "zIndex": 2,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "fontSize": "14px",
                      "marginTop": "-20px",
                    }
                  }
                >
                  100
                </div>
              </div>
            </div>
            <div
              className="slider-tracks"
            >
              <div
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": 5,
                    "cursor": "pointer",
                    "height": "10px",
                    "left": "0%",
                    "marginTop": "18px",
                    "position": "absolute",
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Age Range
          </label>
          <div
            className="signup-slider"
            style={
              Object {
                "flexBasis": "calc(100% - 20px)",
                "height": "45px",
                "marginLeft": "10px",
                "marginRight": "10px",
                "position": "relative",
              }
            }
          >
            <div
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "backgroundColor": "rgb(150, 150, 150)",
                  "borderRadius": "5px",
                  "height": "10px",
                  "marginTop": "18px",
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <div
              className="slider-handles"
            >
              <div
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": "50%",
                    "color": "#333",
                    "cursor": "pointer",
                    "height": "20px",
                    "left": "0%",
                    "marginLeft": "-10px",
                    "marginTop": "12px",
                    "position": "absolute",
                    "textAlign": "center",
                    "width": "20px",
                    "zIndex": 2,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "fontSize": "14px",
                      "marginTop": "-20px",
                    }
                  }
                >
                  18
                </div>
              </div>
              <div
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": "50%",
                    "color": "#333",
                    "cursor": "pointer",
                    "height": "20px",
                    "left": "100%",
                    "marginLeft": "-10px",
                    "marginTop": "12px",
                    "position": "absolute",
                    "textAlign": "center",
                    "width": "20px",
                    "zIndex": 2,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "fontSize": "14px",
                      "marginTop": "-20px",
                    }
                  }
                >
                  100
                </div>
              </div>
            </div>
            <div
              className="slider-tracks"
            >
              <div
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#2185d0",
                    "borderRadius": 5,
                    "cursor": "pointer",
                    "height": "10px",
                    "left": "0%",
                    "marginTop": "18px",
                    "position": "absolute",
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="ui grid"
        >
          <div
            className="center aligned column"
          >
            <button
              className="sc-dkPtRN gFvYMU"
            >
              SIGN UP
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
